src/multio/action/encode/GribEncoder.cc

```diff
@@ -360,13 +360,25 @@ void GribEncoder::setOceanMetadata(const message::Message& msg) {
     }
 
     // Set ocean grid information
-    setValue("unstructuredGridType", config_.getString("grid-type"));
+    if ( config_.has("grid-type") ){
```

Comment:
We could also have the one-liner:
```c++
setValue("unstructuredGridType", config_.getString("grid-type", "undefined"));
```
That would set `grid-type` to `undefined` automatically if that key is not found in the configuration.